spec/javascripts/banksy_spec.js

```diff
@@ -1,42 +1,280 @@
 
 describe("Banksy", function() {
 
   beforeAll(function() {
 
-    editor = woofmark($('#woofmark')[0], {
+    el = $('#woofmark')[0];
+
+    editor = woofmark(el, {
       parseMarkdown: megamark,
       parseHTML:     domador
     });
 
-  });
-
-
-  it("initializes", function() {
-
-    var horse = horsey(editor.editable, {
+    horse = horsey(el, {
       anchor: '@',
       suggestions: [
-        '@hodor'
-        //{ value: '@hodor',  text: 'Hodor'   },
+        { value: '@hodor',  text: 'Hodor' }
       ],
       set: function (value) {
         if (wysiwyg.mode === 'wysiwyg') {
-          editor.textarea.innerHTML = value;
+          editor.editable.innerHTML = value;
         } else {
           editor.textarea.value = value;
         }
       }
     });
 
-    var bridge = banksy(editor.textarea, {
+    bridge = banksy(el, {
       editor: editor,
       horse: horse
     });
 
+    enter = function(before, after) {
+
+      before = before || "";
+      after  = after  || "";
+
+      editor.runCommand(function(chunks, mode) {
+        chunks.before = chunks.before + before;
+        chunks.after = after + chunks.after;
+      });
+ 
+      // trigger the horsey:
+      crossvent.fabricate(el, "keypress");
+
+    }
+
+  });
+
+
+  it("initializes", function() {
+
     expect($('.sey-list')).not.toBe(null);
+    expect($('.sey-list .sey-item').html()).toBe("Hodor");
+
+  });
+
+
+  it("displays on @ token entry", function(done) {
+
+    editor.value('');
+    editor.setMode('wysiwyg');
+
+    enter('hello');
+    expect($('.sey-list .sey-item:first').is(':visible')).toBe(false);
+
+    function onShow() {
+
+      expect($('.sey-list .sey-item:first').is(':visible')).toBe(true);
+      expect(editor.value()).toEqual('hello @');
+
+      crossvent.remove(editor.editable, 'horsey-show', onShow);
+
+      done();
+
+    }
+
+    enter(' @');
+
+    crossvent.add(editor.editable, 'horsey-show', onShow);
+
+  });
+
+
+  it("autocompletes on selection click in wysiwyg mode", function(done) {
+
+    editor.setMode('wysiwyg');
+    editor.value('');
+
+    function onHide() {
+
+      expect($('.sey-list .sey-item:first').is(':visible')).toBe(false);
+
+      crossvent.add(editor.editable, 'horsey-show', onShow);
+
+      enter(' @');
+
+    }
+ 
+    function onShow() {
+
+      // Somehow, dropping priority gives time for this to run; 
+      // possibly for the element to appear and/or move?
+      setTimeout(function() {
+
+        expect($('.sey-list .sey-item:first').is(':visible')).toBe(true);
+        expect(editor.value()).toEqual('hello wysiwyg @');
+ 
+        crossvent.add(editor.editable, 'horsey-selected', onSelect);
+ 
+        crossvent.fabricate($('.sey-list .sey-item:first')[0], "click");
+
+      }, 0);
+
+    }
+ 
+    function onSelect() {
+
+      expect(editor.value()).toEqual('hello wysiwyg @hodor');
+  
+      editor.runCommand(function(chunks, mode) {
+console.log(chunks,"expect insertion point to be at the end");
```

Comment:
@bevacqua - I'm circling back after a few weeks of other work, and I've tracked things to here. Using `console.log` I can see that `chunks.before` and `after` are correct within banksy's custom `appendHTML()` function. But by the time I test them here (triggered by the `crossvent.fabricate` on line 111), the insertion point has moved to the beginning of the text, and `chunks.before` and `after` have changed. Any idea on when/where/how this might be happening? Maybe some kind of focus/refocus event in the midst of user interaction? Thanks!